crypto/evp/ctrl_params_translate.c

```diff
@@ -2845,7 +2847,7 @@ static int evp_pkey_ctx_setget_params_to_ctrl(EVP_PKEY_CTX *pctx,
         fixup_args_fn *fixup = default_fixup_args;
         int ret;
 
-        tmpl.action_type = action_type;
+        tmpl.action_type = ctx.action_type = action_type;
```

Comment:
Could you please elaborate why this is wrong to set here? What could happen otherwise?

Reply:
`evp_pkey_ctx_setget_params_to_ctrl()` is a general function, while setting `ctx.action_type = action_type` is very clearly specific for `ecdh_cofactor`.  The ramifications of making that modification here rather than in the specific fixup function is more than I think we should try to imagine.

Reply:
Hmm... but ctx.action_type will otherwise stay initialized to 0. I do not see how setting the action type on ctx to what action we are actually performing (i.e. SET or GET) can break things.

Reply:
It may be that no action is valid unless the translation or fixup says otherwise.
(valid insofar that it can be used to detect an incorrect call)

Reply:
Hm... OK that makes sense then. @s4ex Could you please replace this and rebase to resolve conflicts?

Reply:
So the general idea is - `if (translation == NULL)`, then `ctx->action_type` should stay `NONE` , but for exceptional cases (i.e. `ecdh_cofactor`, we set it to specific value inside that particular `fixup` function? Is that correct understanding?

Reply:
Yes

Reply:
Sorry for delay. Made the proposed changed and rebased the branch. The commit with previous generic changes is redundant, so removed completely.